Modifies Greeter.java to remove whitespace and @#$% characters and te…

aamanhey opened the pull request:
…st the method.

Commit:
Modifies Greeter.java to remove whitespace and @#$% characters and test the method.

danyag0820 commented on walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java:
```diff
@@ -11,17 +11,23 @@
   /**
    * Returns a greeting for the given name.
    */
+  //private static final String[] invalidArray = new String[] {' ','@','#','$','%'};
```
nit: maybe remove this if it isn't being used anymore?

danyag0820 commented on walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java:
```diff
@@ -11,17 +11,23 @@
+    }
     return "Hello " + name;
   }
 }
```
nit: newline at end

danyag0820 commented on walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:
```diff
@@ -11,23 +11,32 @@
+    // Whitespace should be trimmed
     Assert.assertEquals("Hello Ada", greeting);
   }
 }
```
nit: newline at end

danyag0820 commented on walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:
```diff
@@ -11,23 +11,32 @@
+ 
+    // Whitespace should be trimmed
     Assert.assertEquals("Hello Ada", greeting);
   }
```
optional: maybe add a test for trimming the symbols?

danyag0820 reviewed (commented):
just a few small things :) good work!

Commit:
Adds tester for symbols and adds another method to Greeter to trim from specific list

Commit:
Adds newline

aamanhey commented on walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java:
```diff
@@ -21,10 +21,15 @@
+  //custom trimmer
+  private static final String[] invalidArray = {" ","@","%","!"};
+  
+  public String greetFromSpecificList(String name) {
```
Decided to create a new method to test different methods of implementation.

Commit:
Adds new tests for optional attendees to FindMeetingQueryTest

aamanhey committed:
Delete FindMeetingQueryTest.java

This file is not directly related to the FindMeetingQueryTest.java file and will be placed in another commit.

cjwert-g commented on walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java:
```diff
@@ -11,17 +11,29 @@
+  }
+
+  //custom trimmer
+  private static final String[] invalidArray = {" ","@","%","!"};
```
nit: there appears to be some spacing / formatting issues in this file. I suspect running clang should fix it.

cjwert-g replied:
private static final consts should be named in all-caps

https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

Commit:
Formats with clang

Commit:
Merge branch 'tempTestBranch' of https://github.com/aamanhey/my-portfolio into tempTestBranch

Commit:
Adds the Find Meeting Query test back that was unintentionally deleted with branch merge

aamanhey commented:
There was a merge with this branch that deletes the FindMeetingQuery tester so the subsequent commit was to add the FindMeetingQuery test back. It was already reviewed and merged in a prior pull request.

cjwert-g commented on walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java:
```diff
@@ -21,7 +21,18 @@
+
+  public String greetFromSpecificList(String name) {
+    for (String c : INVALID_ARRAY) {
+      name = name.replaceAll(c, "");
```
If you're not familiar with Big-O complexities, we can discuss this more in one of our stand-ups.

I suspect that replaceAll runs in O(N) time. However we need to call replaceAll for however many elements we have in INVALID_ARRAY. So if we have M elements, we're looking at O(N*M) time. I'm wondering if we could do better than that by concatenating INVALID_ARRAY into a single string and then calling replaceAll just once. 

What do you think?

cjwert-g commented on walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:
```diff
@@ -30,4 +29,29 @@ public void testGreeting() {
+
+    String greeting = greeter.greet("   Ada   ");
+
+    // Whitespace should be trimmed
```
nit: probably don't need this comment since the test method has a descriptive name!

cjwert-g reviewed (approved):
Good work!
